crates/zk-por-core/src/circuit_registry/registry.rs

```diff
@@ -0,0 +1,167 @@
+use plonky2::plonk::{
+    circuit_data::{CircuitConfig, CircuitData},
+    proof::ProofWithPublicInputs,
+};
+
+use crate::{
+    account::Account,
+    // circuit_config::{BATCH_SIZE, RECURSIVE_FACTOR, ASSET_NUM},
+    merkle_sum_prover::{
+        circuits::{
+            account_circuit::AccountTargets, merkle_sum_circuit::build_merkle_sum_tree_circuit,
+        },
+        prover::MerkleSumTreeProver,
+    },
+    recursive_prover::prover::RecursiveProver,
+    recursive_prover::recursive_circuit::{build_recursive_n_circuit, RecursiveTargets},
+};
+
+use plonky2_field::types::Field;
+use rand::Rng;
+
+use crate::types::{C, D, F};
+// use once_cell::sync::Lazy;
+
+pub struct CircuitRegistry<const RECURSIVE_FACTOR: usize> {
+    batch_circuit: (CircuitData<F, C, D>, Vec<AccountTargets>),
+    empty_batch_proof: ProofWithPublicInputs<F, C, D>,
+    recursive_circuits_and_empty_proofs: Vec<(
+        (CircuitData<F, C, D>, RecursiveTargets<RECURSIVE_FACTOR>),
+        ProofWithPublicInputs<F, C, D>,
+    )>,
+}
+
+fn gen_empty_accounts(num_accounts: usize, num_assets: usize) -> (Vec<Account>, u32, u32) {
```

Comment:
that's not really empty account; for circuit building; just fill the vector of account with real empty_account should work